src/software/backend/simulator_backend.h

```diff
@@ -1,35 +1,42 @@
 #pragma once
 
+#include <future>
 #include <mutex>
 
 #include "software/backend/backend.h"
+#include "software/backend/simulation/physics_simulator.h"
 #include "software/world/world.h"
 
 /**
  * This class implements the Backend interface using a physics simulation to
  * execute received commands and provide new data to the system.
- *
- * The simulation uses Box2D as the physics engine, and the abstractions can be found
- * in the `simulation` folder. For more information and examples of how to use Box2D
- * see the following:
- * - https://box2d.org/documentation/
- * - https://github.com/libgdx/libgdx/wiki/Box2d
- * - https://www.iforce2d.net/b2dtut/
  */
 class SimulatorBackend : public Backend
 {
    public:
     static const std::string name;
 
+    /**
+     * Supported modes for simulation speed.
+     *
+     * FAST_SIMULATION will run the simulation as fast a possible, while
+     * REALTIME_SIMULATION will run the simulation in real-time. This is useful
+     * if you want to visualize the simulation.
```

Comment:
Can you define what you mean by "in real-time", it's hard to grasp what speed it will actually run at in a metric/unit sense

Reply:
clarified. let me know if the new comment is better.

Reply:
makes more sense now! thank you